### PR TITLE
driver/at86rf2xx: optimized auto initialization

### PR DESCRIPTION
**Pull request by haukepetersen**
simplified the auto configuration for the Atmel radio driver by introducing a default `at86rf2xx_params.h` file.

This PR is meant as an example how I could imagine all devices drivers in RIOT to be structured in the future... (see #4271 for a description of the proposed concept)

### CODE REVIEW

**Comment by OlegHahm** [comment]
I like.

**Comment by authmillenon** [comment]
Me too :-)

**Comment by cgundogan** [comment]
Seems reasonable. Reduces a lot of code duplication.

**Comment by haukepetersen** [comment]
so anyone bold enough to ack this? @kaspar030?

**Comment by kaspar030** [comment]
I like a lot that at86rf2xx_params.h exists only once.

Still I'd prefer having the configuration in C structs. The whole point of introducing *_param_t style structs was to end up with iterable, type checked configuration data instead of error-prone and clumsy defines.

What do you thing about adopting the shared at86rf_params_t from this PR, but put something like this in ```periph_conf.h``` (per board):

```
#define HAS_ONBOARD_AT86RF2XX

static const at86rf2xx_params_t _at86rf2xx_onboard = {
            .spi = SPI0,
            .spi_speed = blah,
            .cs_pin = blubb,
            .int_pin = pin,
            .sleep_pin = xx,
            .reset_pin = xx,
        };
```
(not an array, just one at86rf2xx_params_t)

and conditionally add that to the default ```at86rf2xx_params.h``` array?
Then later we can also include e.g., a "periph_conf_application.h" and add more conditional entries later.

**Comment by haukepetersen** [comment]
changed the default config file slightly again - the default board configuration is now given as a single define. This has the advantage, that it can be simply re-used in any overriding `xx_params.h` file...

@kaspar030: is this what you had in mind?

**Comment by kaspar030** [comment]
Yes!

If you change the definitions in board.h to { .spi=blah, .spi_speed=blah ...}, the compiler has even more information to detect errors.

**Comment by haukepetersen** [comment]
true, also nicer to read. Will do

**Comment by haukepetersen** [comment]
done.

File: drivers/at86rf2xx/include/at86rf2xx_params.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#define AT86RF2XX_PARAM_INT         (GPIO_PIN(0, 1))
+#endif
+#ifndef AT86RF2XX_PARAM_SLEEP
+#define AT86RF2XX_PARAM_SLEEP       (GPIO_PIN(0, 2))
+#endif
+#ifndef AT86RF2XX_PARAM_RESET
+#define AT86RF2XX_PARAM_RESET       (GPIO_PIN(0, 3))
+#endif
+
+#define AT86RF2XX_PARAMS_DEFAULT    {AT86RF2XX_PARAM_SPI, \
```
Comment (kaspar030) [inline drivers/at86rf2xx/include/at86rf2xx_params.h]: you missed these... :)

**Comment by kaspar030** [comment]
I like, please squash!

File: drivers/at86rf2xx/include/at86rf2xx_params.h (hunk 1 above)
Comment (haukepetersen) [inline drivers/at86rf2xx/include/at86rf2xx_params.h]: ups

**Comment by haukepetersen** [comment]
fixed and squashed.

**Comment by OlegHahm** [comment]
Ready for merging? @kaspar030 you acked, right?

**Comment by haukepetersen** [comment]
not explicitly :-)

**Comment by kaspar030** [comment]
I only have one board with the chip, so I can't actually test the radio... ACK if someone can confirm it's still working.

**Comment by OlegHahm** [comment]
Hm, seems to break something on iotlab-m3. `ifconfig` doesn't show any output any more.

**Comment by haukepetersen** [comment]
ups, breaking through rebasing... I'll look into it

**Comment by haukepetersen** [comment]
fixed (and squashed): the sleep and reset pins were switched around for the `fox` and the `iotlab`. Also fixed a missing newline in the auto-init/at86rf2xx file..

**Comment by OlegHahm** [comment]
Looks good. ACK and go as soon as Travis agrees.

**Pull request merged**